docs/runtime_image.md

```diff
@@ -0,0 +1,83 @@
+# Runtime image
+
+## Overview
+
+For a dynamic languages like PHP, Python or Ruby, the build-time and run-time environments are the same. In this case using builder as a base image for a resulting image is natural.
+
+For compiled languages like C/C++, Java or Golang, the dependencies necessary for compilation might dramatically outweigh the actual runtime artifacts. To keep runtime images slim, S2I enables multiple-step build processes, where a binary artifact such as an executable or Java WAR file is created in the first builder image, extracted, and injected into an second image that simply places the executable in the correct location for execution. To give you even more abilities to customize a resulting image, S2I is also executing `assemble-runtime` script inside of this (runtime) image. In this way you may do a final adjustments by modifying files before an image will be committed.
```

Comment:
"might dramatically outweigh the size of the actual runtime artifacts, or provide attack surface areas that are undesirable in an application image."